src/robotFactory.js

```diff
@@ -1,10 +1,102 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(
+    name,
+    weight,
+    {
+      x = 0,
+      y = 0,
+    },
+    chipVersion
+  ) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x,
+      y,
+    };
 
-class DeliveryDrone {}
+    this.chipVersion = chipVersion;
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+
+    return this;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+
+    return this;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+
+    return this;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+
+    return this;
+  }
+
+  getInfo() {
+    return `Robot: ${this.name}, `
+    + `Chip version: ${this.chipVersion}, `
+    + `Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+
+    this.coords.z = coords.z || 0;
+  }
+
+  goUp(step = 1) {
+    this.coords.z += step;
+
+    return this;
+  }
+
+  goDown(step = 1) {
+    this.coords.z -= step;
+
+    return this;
+  }
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(
+    name,
+    weight,
+    coords,
+    chipVersion,
+    maxLoadWeight,
+    currentLoad = null
+  ) {
+    super(name, weight, coords, chipVersion);
+
+    this.maxLoadWeight = maxLoadWeight;
+    this.currentLoad = currentLoad;
+  }
+
+  hookLoad(cargo) {
+    if (this.currentLoad === null && this.maxLoadWeight >= cargo.weight) {
```

Comment:
better to check for the falseness instead
```suggestion
    if (!this.currentLoad&& this.maxLoadWeight >= cargo.weight) {
```